### ISSUE TITLE
Some results on different reward models

### ISSUE THREAD

**Issue by xuyxu** [XueZeyue/DanceGRPO#36]
Hi, thanks for this great work. We would like to share some further results on different reward models.

* PickScore is an effective reward model for enhancing image aesthetic quality, while preserving realism better than HPSv2. Below are images generated using the original FLUX.1-dev model, as well as versions fine-tuned with DanceGRPO after 100 and 200 training steps.

<img width="1102" height="1172" alt="Image" src="https://github.com/user-attachments/assets/d9c58a9b-1073-4bad-be72-9e70056ea3d2" />

* ImageReward is not a good reward model, leading to corrupted images.

<img width="1098" height="1186" alt="Image" src="https://github.com/user-attachments/assets/a5bcefa6-9522-4d1b-aaa0-010d8bd0b855" />

* For a comparison, below are the results of using HPSv2 as the reward model.

<img width="1096" height="1164" alt="Image" src="https://github.com/user-attachments/assets/8a794c95-d541-4fd5-9c64-d0e9cd783480" />

* Setting the sample image width or height to 720 or 768 can result in "grid patterns" when generating 1024-sized images using the fine-tuned model, increasing the sample image size directly to 1024 during GRPO helps avoid this issue.
* A learning rate of `1e-5` tends to be slightly too high and may degrade image details after a certain number of training steps. Using a smaller value, such as `5e-6`, can help preserve finer details.

**Comment by XueZeyue** [comment]
Many thanks for your results!!!

Actually, I also have many "grid patterns" in my internal experiments if I continue to use my iterations to train (to improve the generalization across prompts). Increasing the image size can improve the results, but it cannot solve the issue. I have also had several discussions with my friends, and someone suggested that it might be due to VAE. I always use EMA to avoid the "grid pattern" in my internal experiments, but EMA also weakens the results of the reward model.

I believe solving the "grid pattern" is a very significant research problem for the future.

**Comment by xuyxu** [comment]
Based on my practice, "grid pattern" is possibly related to the inappropriate perception of attention modules on the sequence length and ROPE. This problem is most obvious if using FLUX.1-dev to generate extremely high resolution images (e.g., 2560 * 2560).

You could check these posts :-)

* https://www.reddit.com/r/StableDiffusion/comments/1g7b76e/a_possible_fix_for_the_gridlines_effect_when/
* https://www.reddit.com/r/StableDiffusion/comments/1eq8zzb/flux_img2img_artifacts_grid_pattern_how_to_solve/

**Comment by WuTao-CS** [comment]
Hi, would you mind open-sourcing the training reward curve of pickscore? It would be useful as a reference for reproducing the same results.

**Comment by xuyxu** [comment]
> Hi, would you mind open-sourcing the training reward curve of pickscore? It would be useful as a reference for reproducing the same results.

Of course. No time to figure out what happened after train step 200. Maybe your results could be helpful after comparison

<img width="1252" height="922" alt="Image" src="https://github.com/user-attachments/assets/fa9478a3-43b6-4548-bcab-bf1b1091ad14" />

**Comment by WuTao-CS** [comment]
> f course. No time to figure out what happened after train

Thank you so much!

**Comment by peki12345** [comment]
@xuyxu Hi, could you share the prompts used to generate these images?

**Comment by XueZeyue** [comment]
For reward collapse (of 8 GPUs), you can refer to this [issue](https://github.com/XueZeyue/DanceGRPO/issues/55).
Maybe it's because the grad norm boom.

**Comment by TianyuCuiOvO** [comment]
> Many thanks for your results!!!
> 
> Actually, I also have many "grid patterns" in my internal experiments if I continue to use more iterations to train (to improve the generalization across prompts). Increasing the image size can improve the results, but it cannot solve the issue. I had several discussions with my friends, and someone suggested that it might be due to VAE. I always use EMA to avoid the "grid pattern" in my internal experiments, but EMA also weakens the results of the reward model.
> 
> I believe solving the "grid pattern" is a very significant research problem for the future.

@XueZeyue still wondering have u found any other solutions to the grid patterns? This effect also bothers me too

**Comment by xuyxu** [comment]
> > Many thanks for your results!!!
> > 
> > Actually, I also have many "grid patterns" in my internal experiments if I continue to use more iterations to train (to improve the generalization across prompts). Increasing the image size can improve the results, but it cannot solve the issue. I had several discussions with my friends, and someone suggested that it might be due to VAE. I always use EMA to avoid the "grid pattern" in my internal experiments, but EMA also weakens the results of the reward model.
> > 
> > I believe solving the "grid pattern" is a very significant research problem for the future.
> 
> @XueZeyue still wondering have u found any other solutions to the grid patterns? This effect also bothers me too

@TianyuCuiOvO Using dynamic sample resolution per group seems a good solution

**Comment by AveryLiu** [comment]
In my internal experiment, I also encountered the 'grid issue'. The moving average denormed vq reward seems good and increases to approximately 7. I guess this is related to reward hacking and stronger reward model shall be applied.
